### ISSUE TITLE
Support for clip/cull planes in ES: EXT_clip_cull_distance

### ISSUE THREAD

**Issue by pknowlesnv** [KhronosGroup/glslang#566]
EXT_clip_cull_distance exposts ARB_cull_distance to GL ES.
https://www.khronos.org/registry/gles/extensions/EXT/EXT_clip_cull_distance.txt

It looks like the version/profile requirements in Initialize.cpp will need updating to include ES for the following:
gl_ClipDistance
gl_CullDistance
gl_MaxClipDistances
gl_MaxCullDistances
gl_MaxCombinedClipAndCullDistances

Also just to double check, the version number requirements for cull distance are 450, but only OpenGL 3.0 is required. Is this because a `#extension` directive is needed to enable them before `#version 450`?

**Comment by johnkslang** [comment]
Yes, if this extension gets implemented, it would change what versions trigger these variables.

In glslang, I only see `gl_CullDistance[]` added as a built-in variable when `version >= 450`, which seems correct.

**Comment by strandborg** [comment]
We at Unity would like to see this get implemented soon, as AFAIK the only way to get glslang to produce RelaxedPrecision stuff is to use ES profile, but that in turn will disable all user clip planes. Having this be controlled by an extension would also be a good way to control whether the SPIR-V shader output requires the ClipDistance capability.

**Comment by johnkslang** [comment]
> the only way to get glslang to produce RelaxedPrecision stuff is to use ES profile

This actually is not true.  When generating SPIR-V, GLSL **mediump** and **lowp** translate to SPIR-V **RelaxedPrecision**.
